File: source/spring-petclinic-rest/src/main/java/com/spring_petclinic/spring_petclinic_rest/data/SpecialtyRepository.java
package com.spring_petclinic.spring_petclinic_rest.data;

import org.springframework.data.jpa.repository.JpaRepository;
import com.spring_petclinic.spring_petclinic_rest.domain.models.Specialty;
import com.spring_petclinic.spring_petclinic_rest.intent.IntentMerge;

/**
 * Spring Data JPA repository for the Specialty entity.
 */
@IntentMerge
public interface SpecialtyRepository extends JpaRepository<Specialty, Integer> {
}

File: source/nestjs-petclinic-rest/src/repository/specialty.repository.ts
import { Repository, EntityRepository } from 'typeorm';
import { CustomRepository } from './../typeorm/typeorm-ex.decorator';
import { Specialty } from './../domain/entities/specialty.entity';
import { IntentMerge } from './../intent/intent.decorators';

@IntentMerge()
@CustomRepository(Specialty)
export class SpecialtyRepository extends Repository<Specialty> {
}